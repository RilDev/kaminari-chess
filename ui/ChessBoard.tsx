import { chessBoardLayout } from "@/data";

export default function ChessBoard() {
  return (
    <div className="border border-black">
      {chessBoardLayout.map((row, rowIndex) => (
        <div key={rowIndex} className="flex flex-row">
          {row.map((square, squareIndex) => (
            <div
              key={squareIndex}
              className={`w-24 h-24 ${
                square === "white" ? "bg-white" : "bg-black"
              }`}
              data-testid="square"
            />
          ))}
        </div>
      ))}
    </div>
  );
}
