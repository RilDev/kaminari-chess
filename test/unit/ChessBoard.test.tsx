import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import ChessBoard from "@/ui/ChessBoard";

describe("ChessBoard", () => {
  it("renders 16 squares", () => {
    render(<ChessBoard />);
    expect(screen.getAllByTestId("square").length).toBe(16);
  });

  it("renders chess board unchanged", () => {
    const { container } = render(<ChessBoard />);
    expect(container).toMatchSnapshot();
  });
});
