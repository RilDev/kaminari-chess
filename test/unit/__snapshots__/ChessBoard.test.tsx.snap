// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChessBoard renders chess board unchanged 1`] = `
<div>
  <div
    class="border border-black"
  >
    <div
      class="flex flex-row"
    >
      <div
        class="w-24 h-24 bg-white"
        data-testid="square"
      />
      <div
        class="w-24 h-24 bg-black"
        data-testid="square"
      />
      <div
        class="w-24 h-24 bg-white"
        data-testid="square"
      />
      <div
        class="w-24 h-24 bg-black"
        data-testid="square"
      />
    </div>
    <div
      class="flex flex-row"
    >
      <div
        class="w-24 h-24 bg-black"
        data-testid="square"
      />
      <div
        class="w-24 h-24 bg-white"
        data-testid="square"
      />
      <div
        class="w-24 h-24 bg-black"
        data-testid="square"
      />
      <div
        class="w-24 h-24 bg-white"
        data-testid="square"
      />
    </div>
    <div
      class="flex flex-row"
    >
      <div
        class="w-24 h-24 bg-white"
        data-testid="square"
      />
      <div
        class="w-24 h-24 bg-black"
        data-testid="square"
      />
      <div
        class="w-24 h-24 bg-white"
        data-testid="square"
      />
      <div
        class="w-24 h-24 bg-black"
        data-testid="square"
      />
    </div>
    <div
      class="flex flex-row"
    >
      <div
        class="w-24 h-24 bg-black"
        data-testid="square"
      />
      <div
        class="w-24 h-24 bg-white"
        data-testid="square"
      />
      <div
        class="w-24 h-24 bg-black"
        data-testid="square"
      />
      <div
        class="w-24 h-24 bg-white"
        data-testid="square"
      />
    </div>
  </div>
</div>
`;
